File: test/Test_Ekado.cpp
#include "../src/Ekado.cpp"
#include <gtest/gtest.h>

using namespace std;

TEST(EkadoTest, AllFunctionalityTest){
    Ekado testEkado(10);
    LinkedList<FarmProductType> EkadoRecipe = testEkado.getResep();
    ASSERT_EQ(EGG, EkadoRecipe.get(0));
    ASSERT_EQ(MEAT, EkadoRecipe.get(1));
    ASSERT_EQ(EGG, testEkado.getResepwithIdx(0));
    ASSERT_EQ(MEAT, testEkado.getResepwithIdx(1));
    ASSERT_EQ(10, testEkado.getPrice());
    ASSERT_EQ(EKADO, testEkado.getObjectType());
    ASSERT_EQ(SIDEPRODUCT, testEkado.getProductType());
}

File: Java/src/test/java/engisubs/gameobject/product/sideproduct/TestEkado.java
package engisubs.gameobject.product.sideproduct;

import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;

import engisubs.gameobject.GameObject.FarmProductType;
import engisubs.gameobject.GameObject.GameObjectType;
import engisubs.gameobject.product.Product.ProductType;

public class TestEkado{
    private Ekado test;

    @Before
    public void setUpObject() {
        test = new Ekado(100);
    }
    @Test
    public void instantiationTest(){
        Assert.assertEquals(100, test.getPrice());
        Assert.assertEquals(GameObjectType.EKADO, test.getGameObjectType());
        Assert.assertEquals(ProductType.SIDEPRODUCT, test.getProductType());
        Assert.assertEquals("Tiongkok's Special Ekado~~", test.getProductName());
    }
    @Test
    public void resepTest(){
        Ekado.InitResep();
        Assert.assertArrayEquals(new FarmProductType[]{FarmProductType.EGG, FarmProductType.MEAT}, Ekado.getResep().toArray());
        Assert.assertEquals(null, Ekado.getResepwithIdx(3));
    }
}